### PR TITLE
Add markdown support to lecture and post comments

### PR DESCRIPTION
**Pull request by gabrielKerekes**
Coded on stream.

Could be made a bit nicer for sure, but IMO this is good enough for now.

I added a task to Trello for refactoring the whole comments stuff because there is a lot of duplication.

Edditing with markdown:
image

Markdown preview:
image

### CODE REVIEW

File: src/components/domain/lecture-comment/EditableLectureComment.tsx [hunk 1]
```diff
@@ -33,14 +37,48 @@ const EditableLectureComment = ({
     })
   }
 
+  const togglePreview = (): void => {
+    setIsInPreview(!isInPreview)
+  }
+
+  const containsMarkdown =
```
Comment (gabrielKerekes) [inline src/components/domain/lecture-comment/EditableLectureComment.tsx:44]: We can (and probably should) improve this later on.

**Comment by xjahic** [comment]
The current implementation contains some bugs and needs improvement.
Screenshot 2023-03-21 193812
Bug 1. - If you submit in preview mode, the preview mode icon will stay in put. 
Bug 2. - The icon should not be shown if text is empty
Bug 3. - Weird scrollbar in comments. 

About improvements. 
- The icon should be accompanied by some text that could also signal user that using markdown is an option. 
- Some padding

You can see my idea of improvements in this 2 screenshots
Screenshot 2023-03-21 195628

Screenshot 2023-03-21 195645
Code version - https://github.com/StreetOfCode/streetofcode-web/tree/jj/markdown-in-comments-v1

I have also tried using MDEditor to see if it was customizable and I think we could easily work with it if we wanted. We would set some good default values and remove few toolbar options and be good to go. Also screenshot.
Screenshot 2023-03-21 201730
Code version - https://github.com/StreetOfCode/streetofcode-web/tree/jj/makrdown-with-mdeditor

**Review by xjahic — CHANGES_REQUESTED** [review]
The current implementation contains some bugs and needs improvement.
Screenshot 2023-03-21 193812
Bug 1. - If you submit in preview mode, the preview mode icon will stay in put. 
Bug 2. - The icon should not be shown if text is empty
Bug 3. - Weird scrollbar in comments. 

About improvements. 
- The icon should be accompanied by some text that could also signal user that using markdown is an option. 
- Some padding

You can see my idea of improvements in this 2 screenshots
Screenshot 2023-03-21 195628

Screenshot 2023-03-21 195645
Code version - https://github.com/StreetOfCode/streetofcode-web/tree/jj/markdown-in-comments-v1

I have also tried using MDEditor to see if it was customizable and I think we could easily work with it if we wanted. We would set some good default values and remove few toolbar options and be good to go. Also screenshot.
Screenshot 2023-03-21 201730
Code version - https://github.com/StreetOfCode/streetofcode-web/tree/jj/makrdown-with-mdeditor

**Comment by gabrielKerekes** [comment]
I feel like md editor is a bit of an overkill. The comments should still be just comments and md-editor is a bit overwhelming for that. My assumption is that 95% of comments will not contain any markdown and so always showing md-editor isn't nice. I also wasn't able to quickly make it work with darkmode for example. For now, I'd leave this as it is in this PR. Also when editing a comment the window is just too narrow to be split in half. All of this can of course be addressed I just don't think it's worth it now.

The bugs and improvements you mentioned have been implemented in https://github.com/StreetOfCode/streetofcode-web/pull/44/commits/ed6b6ecd2135a968bfb9ffa6d81347d5a65db289.

**Review by xjahic — APPROVED** [review]
LGTM. Agree on unnecessary md-editor.

**Pull request merged**